WordPress/gutenberg
# Edit/create shadows in global styles

## What?

This change introduces a new item `shadows` in the global styles for **editing/creating shadows**

<img width="285" alt="image" src="https://github.com/WordPress/gutenberg/assets/1935113/1c9a6a9a-0e13-4f95-860e-87618df72ed3">
<img width="286" alt="image" src="https://github.com/WordPress/gutenberg/assets/1935113/0381da3e-60b4-4758-9f6f-a6e26c7482e6">
<img width="841" alt="image" src="https://github.com/WordPress/gutenberg/assets/1935113/c002f1c2-1b39-4188-8fc6-432cd2e9fcee">

## Testing instructions

1. Find newly added item `shadows` in global styles panel
2. Click to land on shadows panel, it should show list of `core`, `theme` and `custom` shadows.
3. If there are not `theme` or `custom` shadows, it should show a placeholder label.
4. Select a shadow to land on preview screen

Editing a shadow:

1. In the preview screen, click on `Drop shadow` to invoke editor, change lengths, color, inset etc.. and preview should update instantly.
2. Click `+` to add new shadow layer.
3. click `-` to remove a shadow layer. (this is available when there are at least 2 layers)
4. Save the edits.

## Subtasks

- [x] Show list of `default`, `theme` and `custom` shadows.
- [x] Preview shadow
- [x] Update a shadow (editor panel)
- [x] Add a shadow
- [x] Delete a shadow
- [ ] Duplicate a shadow
- [x] Rename a shadow (how about slug??, also handle conflicts)
- [x] add or delete a shadow layer (in second screen)

## Related issues:

fixes: #57100
